src/main/java/org/reaktivity/nukleus/tls/internal/stream/TlsClientFactory.java

```diff
@@ -685,7 +690,7 @@ private int decodeHandshakeNeedUnwrap(
         int limit)
     {
         final int length = limit - progress;
-        if (length != 0)
+        if (length >= 0)
```

Comment:
Can this be optimized like server case to ensure handshake has not completed to allow `== 0`, perhaps by checking `tlsStream` state if always present, or else just `!tlsStream.isPresent()` if deferred?

Reply:
yes it can be optimized thanks!